subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/Transformer.java

```diff
@@ -47,4 +48,6 @@
      * This includes the parameters and the implementation.
      */
     HashCode getSecondaryInputHash();
+
+    void isolateParameters();
```

Comment:
Can we implement this without modifying internal state, and instead returning an `IsolatedTransformer` or `TransformerWithIsolatedParameters` or something?

Reply:
We always need something which holds the reference from the registration to the isolated transformer. Currently, this is the transformer itself. We could move it a little bit outwards to the `TransformationStep`, making the `TransformationStep` mutable. I don't see this as an improvement, though. WDYT?